OpenRA.Mods.Common/Traits/Render/WithDamageOverlay.cs

```diff
@@ -0,0 +1,74 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2016 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using OpenRA.Graphics;
+using OpenRA.Mods.Common.Warheads;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits.Render
+{
+	[Desc("Renders an overlay when the actor is taking heavy damage.")]
+	public class WithDamageOverlayInfo : ITraitInfo, Requires<RenderSpritesInfo>
+	{
+		public readonly string Image = "smoke_m";
+
+		[SequenceReference("Image")] public readonly string IdleSequence = "idle";
```

Comment:
Technically it is Start + Loop. Somehow inherited from the existing implementation.